tests/test_unit/test_kinematics.py

```diff
@@ -1,112 +1,176 @@
-from contextlib import nullcontext as does_not_raise
-
 import numpy as np
 import pytest
-import xarray as xr
 
 from movement.analysis import kinematics
 
 
-class TestKinematics:
-    """Test suite for the kinematics module."""
+@pytest.mark.parametrize(
+    "valid_dataset_uniform_linear_motion",
+    [
+        "valid_poses_dataset_uniform_linear_motion",
+        "valid_bboxes_dataset",
+    ],
+)
+@pytest.mark.parametrize(
+    "kinematic_variable, expected_2D_array_per_individual_and_kpt",
+    [
+        (
+            "displacement",
+            {
+                0: np.vstack(
+                    [np.zeros((1, 2)), np.ones((9, 2))]
+                ),  # at t=0 displacement is (0,0)
+                1: np.multiply(
+                    np.vstack([np.zeros((1, 2)), np.ones((9, 2))]),
+                    np.array([1, -1]),
+                ),
+            },
+        ),
+        (
+            "velocity",
+            {
+                0: np.ones((10, 2)),
+                1: np.multiply(np.ones((10, 2)), np.array([1, -1])),
+            },
+        ),
+        (
+            "acceleration",
+            {
+                0: np.zeros((10, 2)),
+                1: np.zeros((10, 2)),
+            },
+        ),
+    ],
+)
+def test_kinematics_uniform_linear_motion(
```

Comment:
This test can be simplified as follows, so we don't need multiple asserts in a for loop:

```python
@pytest.mark.parametrize(
    "valid_dataset_uniform_linear_motion",
    [
        "valid_poses_dataset_uniform_linear_motion",
        "valid_bboxes_dataset",
    ],
)
@pytest.mark.parametrize(
    "kinematic_variable, expected_arrays",
    [
        (
            "displacement",
            [
                np.vstack([np.zeros((1, 2)), np.ones((9, 2))]),  # Individual 0
                np.multiply(
                    np.vstack([np.zeros((1, 2)), np.ones((9, 2))]),
                    np.array([1, -1]),
                ),  # Individual 1
            ],
        ),
        (
            "velocity",
            [
                np.ones((10, 2)),  # Individual 0
                np.multiply(
                    np.ones((10, 2)), np.array([1, -1])
                ),  # Individual 1
            ],
        ),
        (
            "acceleration",
            [
                np.zeros((10, 2)),  # Individual 0
                np.zeros((10, 2)),  # Individual 1
            ],
        ),
    ],
)
def test_kinematics_uniform_linear_motion(
    valid_dataset_uniform_linear_motion,
    kinematic_variable,
    expected_arrays,  # 2D: n_frames, n_space_dims
    request,
):
    """Test computed kinematics for a uniform linear motion case.

    Uniform linear motion means the individuals move along a line
    at constant velocity.

    We consider 2 individuals ("id_0" and "id_1"),
    tracked for 10 frames, along x and y:
    - id_0 moves along x=y line from the origin
    - id_1 moves along x=-y line from the origin
    - they both move one unit (pixel) along each axis in each frame

    If the dataset is a poses dataset, we consider 3 keypoints per individual
    (centroid, left, right), that are always in front of the centroid keypoint
    at 45deg from the trajectory.
    """
    position = request.getfixturevalue(
        valid_dataset_uniform_linear_motion
    ).position
    kinematic_variable = getattr(kinematics, f"compute_{kinematic_variable}")(
        position
    )
    expected_da = xr.DataArray(
        np.stack(expected_arrays, axis=1),  # Stack along the individuals axis
        dims=["time", "individuals", "space"],
    )
    if "keypoints" in position.coords:
        expected_da = expected_da.expand_dims(
            {"keypoints": position.coords["keypoints"].size}
        )
        expected_da = expected_da.transpose(
            "time", "individuals", "keypoints", "space"
        )
    # Compare the values of the kinematic_variable against the expected_da
    np.testing.assert_allclose(kinematic_variable.values, expected_da.values)
```

Reply:
much cleaner and excellent point re the asserts in the loop, thank you!